sgt-back-end/index.js

```diff
@@ -0,0 +1,240 @@
+    const error = { error: 'Score is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (req.body.score < 0 || req.body.score > 100) {
+    const error = { error: 'Score is invalid' };
+    res.status(400).json(error);
+    return;
+  }
+  const body = req.body;
+  const empty = [];
+  for (const x in body) {
+    empty.push(body[x]);
+  }
+  const sql = `
+  insert into "grades" ("name", "course", "score")
+  values ($1, $2, $3)
+    `;
+  db.query(sql, empty)
+
+    .then(result => {
+      const max = `select *
+  from "grades"
+ order by "gradeId" desc
+ limit 1;`;
+      db.query(max)
+        .then(result => {
+          const grade = result.rows[0];
+          res.status(201).json(grade);
+        });
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.delete('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const test = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+  const params = [gradeId];
+  db.query(test, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with gradeId: ${gradeId}`
+        });
+        return;
+      }
+      const sql = `
+  delete from "grades"
+ where "gradeId" = $1
+  `;
+      db.query(sql, params)
+        .then(result => {
+          res.status(204).send();
+        })
+        .catch(err => {
+          console.error(err);
+          res.status(500).json({
+            error: 'An unexpected error occurred.'
+          });
+        });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!req.body.name) {
+    const error = { error: 'Name is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (!req.body.course) {
+    const error = { error: 'Course is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (req.body.score < 0 || req.body.score > 100) {
+    const error = { error: 'Score is invalid' };
+    res.status(400).json(error);
+    return;
+  } else if (!req.body.score && req.body.score !== 0) {
+    const error = { error: 'Score is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    const error = { error: '"gradeId" must be a positive integer' };
+    res.status(400).json(error);
+    return;
+  }
+  const body = req.body;
+  const empty = [];
+  for (const x in body) {
+    empty.push(body[x]);
+  }
+  empty.push(gradeId);
```

Comment:
Same here as above, this won't guarantee the right data is in the array or that it is in the correct order. Explicitly create the params array.